desc/protoparse/parser.go

```diff
@@ -104,6 +104,18 @@ type Parser struct {
 	// with import statements all being relative to the root of this tree).
 	InferImportPaths bool
 
+	// DependencyDescriptors is a map of filenames to file descriptor protos,
+	// which will be consulted when resolving imports. This allows a compiled Go
+	// proto in another Go module to be referenced in the proto(s) being parsed.
```

Comment:
I find this wording confusing since this is not the use case I really had in mind. I was more thinking that you might be doing incremental compiles, and already have descriptors for some of the new files' dependencies.

If this is what you are really trying to accomplish, perhaps this should instead be a `func(string) (*desc.FileDescriptor, error)`, so the caller does not have to enumerate all of the provided files up-front. Then you could set this to `desc.LoadFileDescriptor` and not need to create a map. A map could still be created to back it, but then the user would supply a custom function that queries the map.

Reply:
I think we are serving different use cases (which is fine, as long as we can accomplish both of them well!).

I like the idea of `func(string) (*desc.FileDescriptor, error)`. Does that accomplish your use case also? (I think it does.)

Reply:
Done.